helsingborg-stad/app-mitt-helsingborg
# Feat/New case status+period handling

_! Awaiting changes in AWS (https://github.com/helsingborg-stad/helsingborg-io-sls-api/pull/142) before merging to dev !_

## Explain the changes you’ve made

Changed case overview screen to only get cases from the db and removed ability to create new cases from the app. 
Display applying period on not started cases and also on case detail screen.
Changed handling of statuses, each case will have a status object(with type, name and description) instead of a string. 

## Explain your solution

We discussed how we wanted to handle statuses and decided to let AWS handle manipulation of case statuses. But before that actually works I added a temporary mock file with a list of statuses that the app can consume. The app still do manipulation of status in the state and i have **not** changed that logic in this PR. 

 Instead of status strings we add objects that look like this: 
 `{
    type: 'active.ongoing',
    name: 'Pågående',
    description:
      'Du har påbörjat en ansökan. Du kan öppna din ansökan och fortsätta där du slutade.'
  },
`
Notice that the type is "dot notated" to make it easier for us to group cases with different scopes like **active** and **closed** cases. And the second part is a more specific status. And if we want to be super specific we can add like "active.ekb.ongoing" to have unique status messages for different services.
The new solution for statuses is more of a proof of concept right now and is not implemented yet in the backend. 

## How to test the changes?

Since we don't have the new statuses in our backend you need to manually change status of a case in your db to see that this is working. 

- Change case status to an object like the example above. List of more statuses can be found here: https://github.com/helsingborg-stad/app-mitt-helsingborg/compare/feat/case-overview-handle-periods?expand=1#diff-566954af75cb9b57618e232a35151f5f9e089807783cc571940c011245885677
- Then test if the case overview is showing correct month and status.
- Case detail screen is also updated with period(month) and status. 

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.

## Screenshots (optional)
<img width="405" alt="Screenshot 2021-02-01 at 09 20 02" src="https://user-images.githubusercontent.com/21363149/106432276-c806c880-646e-11eb-92d6-43db1a20b5ee.png">
<img width="390" alt="Screenshot 2021-02-01 at 09 20 14" src="https://user-images.githubusercontent.com/21363149/106432285-cb9a4f80-646e-11eb-9fd8-159a2ccca417.png">

## Anyhting else?

This PR can only be merged to dev when Aws is handling status as objects.
